kinto/core/resource/schema.py

```diff
@@ -150,3 +151,142 @@ class BookmarkSchema(ResourceSchema):
+    """Basic querystring field SchemaNode."""
+
+    missing = colander.drop
+
+    def deserialize(self, cstruct=colander.null):
+        if isinstance(cstruct, six.string_types):
+            cstruct = native_value(cstruct)
+        return super(QueryField, self).deserialize(cstruct)
+
+
+class FieldList(QueryField):
+    """String field representing a list of attributes."""
+
+    schema_type = colander.Sequence
+    error_message = "The value should be a list of comma separated attributes"
+    missing = colander.drop
+    fields = colander.SchemaNode(colander.String(), missing=colander.drop)
+
+    def deserialize(self, cstruct=colander.null):
+        if isinstance(cstruct, six.string_types):
+            cstruct = cstruct.split(',')
+        return super(FieldList, self).deserialize(cstruct)
+
+
+class HeaderQuotedInteger(HeaderField):
+    """Integer between "" used in precondition headers."""
+
+    schema_type = colander.String
+    error_message = "The value should be integer between double quotes"
+    validator = colander.Any(colander.Regex('^"([0-9]+?)"$', msg=error_message),
+                             colander.Regex('\*'))
+
+    def deserialize(self, cstruct=colander.null):
+        param = super(HeaderQuotedInteger, self).deserialize(cstruct)
+        if param is colander.drop or param == '*':
+            return param
+
+        return int(param[1:-1])
+
+
+class HeaderSchema(colander.MappingSchema):
+    """Schema used for validating and deserializing request headers. """
+
+    if_match = HeaderQuotedInteger(name='If-Match', missing=colander.drop)
+    if_none_match = HeaderQuotedInteger(name='If-None-Match', missing=colander.drop)
+    response_behaviour = HeaderField(colander.String(),
+                                     name='Response-Behavior',
+                                     validator=colander.OneOf(['full', 'light', 'diff']),
+                                     missing=colander.drop)
+
+    @staticmethod
+    def schema_type():
+        return colander.Mapping(unknown='preserve')
+
+
+class QuerySchema(colander.MappingSchema):
+    """
+    Schema used for validating and deserializing querystrings. It will include
+    and try to guess the type of unknown fields (filters) on deserialization.
+    """
+
+    _limit = QueryField(colander.Integer())
+    _fields = FieldList()
+    _sort = FieldList()
+    _token = QueryField(colander.String())
+    _since = QueryField(colander.Integer())
+    _to = QueryField(colander.Integer())
+    _before = QueryField(colander.Integer())
+    last_modified = QueryField(colander.Integer())
+
+    @staticmethod
+    def schema_type():
+        return colander.Mapping(unknown='ignore')
+
+    def deserialize(self, cstruct=colander.null):
+        values = {}
+
+        schema_values = super(QuerySchema, self).deserialize(cstruct)
+        if schema_values is colander.drop:
+            return schema_values
+
+        # Deserialize filters
```

Comment:
It doesn't look bad by any means IMHO, and I really like having an explicit, formal schema, so I think this is an improvement.